django/utils/timesince.py

```diff
@@ -69,22 +71,38 @@ def timesince(d, now=None, reversed=False, time_strings=None, depth=2):
             leapdays += 1
     delta -= datetime.timedelta(leapdays)
 
+    # fix months duration to be relative to the months between d and now
+    months = now.month - d.month
+    if months > 0:
+        m_days = (sum(MONTHS_DAYS[d.month - 1 : now.month - 1])) / months
+    else:
+        m_days = (365 - sum(MONTHS_DAYS[now.month - 1 : d.month - 1])) / (12 + months)
+    timesince_chunks_adjusted = list(TIMESINCE_CHUNKS)
+    timesince_chunks_adjusted[1] = (int(m_days * 24 * 60 * 60), "month")
+
     # ignore microseconds
     since = delta.days * 24 * 60 * 60 + delta.seconds
     if since <= 0:
         # d is in the future compared to now, stop processing.
         return avoid_wrapping(time_strings["minute"] % {"num": 0})
-    for i, (seconds, name) in enumerate(TIMESINCE_CHUNKS):
+    for i, (seconds, name) in enumerate(timesince_chunks_adjusted):
         count = since // seconds
         if count != 0:
             break
     else:
         return avoid_wrapping(time_strings["minute"] % {"num": 0})
+
     result = []
     current_depth = 0
-    while i < len(TIMESINCE_CHUNKS) and current_depth < depth:
-        seconds, name = TIMESINCE_CHUNKS[i]
+
+    if months <= 0:
+        months = 12 + months
+
```

Comment:
This could be moved with the above code I think. (The `months` variable is not used in-between ln79 and here.) 🤔

If we can adjust the time chucks and the months, all in one place, with a good comment (or docstring if it can be wrapped in a function) then I think it would be clearer.

Reply:
The reason I put this here is because the variable is not needed for the code block above, so to put it there would be confusing in my opinion. I put it close to the loop where it is used, near the other variables that are set up just before the loop. I can add a comment to make it more clear why it is there and why we still need that variable.